apollo-ios/Sources/Apollo/ApolloClientProtocol.swift

```diff
@@ -39,12 +39,15 @@ public protocol ApolloClientProtocol: AnyObject {
   /// - Parameters:
   ///   - query: The query to fetch.
   ///   - cachePolicy: A cache policy that specifies when results should be fetched from the server or from the local cache.
+  ///   - refetchOnFailedUpdates: Should the watcher perform a network fetch when it's watched
+  ///     objects have changed, but reloading them from the cache fails. Should default to `true`.
   ///   - context: [optional] A context that is being passed through the request chain. Should default to `nil`.
   ///   - callbackQueue: A dispatch queue on which the result handler will be called. Should default to the main queue.
   ///   - resultHandler: [optional] A closure that is called when query results are available or when an error occurs.
   /// - Returns: A query watcher object that can be used to control the watching behavior.
   func watch<Query: GraphQLQuery>(query: Query,
                                   cachePolicy: CachePolicy,
+                                  refetchOnFailedUpdates: Bool,
```

Comment:
Are we considering this a breaking change since it would impact users that have written their own `ApolloClientProtocol` conforming client?

Reply:
Ah that's a great point. I've decided to not add the parameter to `ApolloClientProtocol` since it's pretty minor and fix this with a deprecated function on `ApolloClient`. How do you feel about this fix? 

It's a little awkward that the new parameter isn't on the protocol, but honestly, the whole protocol is kind of a painful solution to a problem that should be fixed by us building a `MockApolloClient` for testing and maybe just making `ApolloClient` `open`. That's work for 2.0 though.

Reply:
> I've decided to not add the parameter to ApolloClientProtocol since it's pretty minor and fix this with a deprecated function on ApolloClient. How do you feel about this fix?

Looks good. I think it's a better solution than breaking existing implementations for something that isn't crucial.

Reply:
@AnthonyMDev @calvincestari 

We have wrapped `ApolloClient`, but are currently accepting an `ApolloClientProtocol`. Though I think we can change that to just `ApolloClient`.

I am realizing how _thin_ `ApolloClient` actually is, and have been considering just foregoing wrapping an `ApolloClient` and `ApolloClientProtocol` at all, and just having our own `NetworkTransport` that wraps `RequestChainNetworkTransport` internally.

Just letting you know, if you feel like you'd like to just keep the `ApolloClient` change out (in another commit to main of course), I am fine just using `GraphQLQueryWatcher` directly if you can be certain that `ApolloClient` can remain as thin as it is now until a 2.0/3.0 level of change.

Reply:
Actually a rescind that statement, `GraphQLQueryWatcher` requires an argument conforming to `ApolloClientProtocol`.